step-capstone/src/components/Trip.js

```diff
@@ -0,0 +1,247 @@
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 3,
+        finalized: false,
+        type: "flight",
+        departureAirport: "BOS",
+        arrivalAirport: "SFO",
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 125,
+        finalized: true,
+        type: "flight",
+        departureAirport: "BOS",
+        arrivalAirport: "SFO",
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 87,
+        finalized: true,
+        type: "flight",
+        departureAirport: "BOS",
+        arrivalAirport: "SFO",
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 231,
+        finalized: true,
+        type: "flight",
+        departureAirport: "BOS",
+        arrivalAirport: "SFO",
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 5425,
+        finalized: true,
+        type: "flight",
+        departureAirport: "BOS",
+        arrivalAirport: "SFO",
+        departureDate: "4:00pm EST",
+        arrivalDate: "7:00pm PST",
+        description: "Additional notes"
+    },
+    {
+        id: 100,
+        finalized: true,
+        type: "hotel",
+        name: "Hotel ZED",
+        location: "London",
+        checkIn: "December 17, 2020 15:30:00",
+        checkOut: "December 20, 2020 11:00:00",
+        description: 'description',
+    }
+]
+
+export default class Trip extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            items: []
+        }
+
+        this.handleRemoveItem = this.handleRemoveItem.bind(this);
+        this.handleEditItem = this.handleEditItem.bind(this);
+        this.handleAddItem = this.handleAddItem.bind(this);
+    }
+
+    fetchData() {
+        return testData;
+    }
+
+    componentDidMount() {
+        let data = this.fetchData();
+        this.setState({
+            items: data
+        })
+    }
+
+    handleRemoveItem(id) {
+        console.log("removing" + id)
+        this.setState({
+            items: this.state.items.filter((item) => item.id !== id)
+        });
+    }
+
+    handleEditItem(id) {
+        console.log("editing " + id)
+    }
+
+    handleAddItem(type, data) {
+        if (data === undefined) {
+            console.log("please enter information")
+        } else {
+            // Add to database here
+            // current code for testing data format etc.
+            switch (type) {
+                case "event":
+                    console.log("adding event");
+                    break;
+                case "hotel":
+                    this.setState({
```

Comment:
This re-paramaterizing seems a little redundant. Why can't you simply pass the values, or validate them and then pass?

Reply:
I originally did it to just add the type attribute to the object, but you're right there must be a less redundant way to do that. But when actually adding to the database we'd have to do something like this for each property of the item.